### ISSUE TITLE
Distributed training on multiple GPUs get stucked.

### ISSUE THREAD

**Issue by NotoCJ** [lyuwenyu/RT-DETR#529]
Hello,
I was training with RT-DETRv2 with multiple GPUs. The training got stucked when a batch of data on a GPU are all background images (i.e., no ground-truth/target boxes), the volatile gpu-util became 100%, and there were no errors or warnings reported. The training was just stucked there and did not go on. In such case, the VFL loss is very small (0.0528), l1 and giou box loss are all 0.  The training was stucked in the scaler.backward or scaler.step (or scaler.update maybe) process. When there are some target boxes in a batch data, the training is fine. By the way, when training with gloo backend, it was stucked in the same occasion, and the following error was reported:
<img width="496" alt="image" src="https://github.com/user-attachments/assets/bdebdd10-7abb-4f64-9504-79f5291bdc42" />

Thanks for attention!

**Comment by lyuwenyu** [comment]
1. You can filter relevant samples before training
2. or try `loss_box = pred_box *= 0` in that case.

**Comment by NotoCJ** [comment]
Thanks. But the loss_bbox is already 0 in such case (or maybe i misunderstand?), and i also want to train with these background images to prevent wrong detections. 
I guess this could be a data mismatch problem between different GPUs, according to the error reported when using gloo backend. I found that when there are no target boxes, the "dn_aux_outputs" in model output will not be defined, then dn related loss items will not be defined. I tried to add these loss items with zero values and no grad. Besides, the find_unused_parameters is set as True when setting DDP mode to prevent DDP waiting for these unused parameters. These operations solved the stucking problem during training in my case. 
I'm not sure this is a common problem for such case in which there are many background images, and also the above solution may not be a accurate or good one cause it increased a little bit training time.

**Comment by bryan-pakulski** [comment]
@NotoCJ  any chance you could post the code changes you made?

I believe I have some similar issues, large dataset with many background images

I've modified the loss values and also set find_unused_parameters to True but I haven't been able to have as much luck as you with actually getting training to work for this dataset on multi-gpu

**Comment by NotoCJ** [comment]
@bryan-pakulski of course, i added the following codes (the "else" part) in the forward() function in rtdetrv2_criterion.py

<img width="384" alt="Image" src="https://github.com/user-attachments/assets/cdb09a6f-2f2b-4a58-910d-6cd1830f4e22" />

Also, the num_layers, i.e., the layer count in decoder, is passed to here as following:
<img width="191" alt="Image" src="https://github.com/user-attachments/assets/a08122e1-31c2-4a65-ac70-9c26632c946c" />
and this parameter can be defined in config file rtdetrv2_r50vd.yml as following:
<img width="243" alt="Image" src="https://github.com/user-attachments/assets/0bc341b0-9bf0-470e-974c-9c63197a023e" />
or maybe the num_layers can be obtained in other ways.

**Comment by bhuvnesh-vcti** [comment]
@NotoCJ Hey I tried adding the suggested code. But still the training is freeze state only also my dataset contains background images. 
@lyuwenyu We want to improve model learning by adding negative images. Can you suggest how to solve this issue with multi-gpu training. With single GPU training is working fine.

Setups I've tried:
8xV100 - Didn't worked
4xT4 - Didn't worked
1xT4 - Worked fine.
